atom/browser/native_window.cc

```diff
@@ -204,6 +204,14 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
 
 void NativeWindow::SetSize(const gfx::Size& size, bool animate) {
   SetBounds(gfx::Rect(GetPosition(), size), animate);
+
+  // In offscreen mode call RenderWidgetHostView's SetSize explicitly
+  const auto view = web_contents()->GetRenderWidgetHostView();
+  if (view) {
```

Comment:
How about subscribing to window resize events in `RenderWidgetHostView`? It is provided by `NativeWindowObserver`.